common/openapi.yml

```diff
@@ -258,8 +258,52 @@ paths:
         '500':
           $ref: '#/components/responses/ServerError'
 
+  /api/policy/v1alpha1/region/{objectId}/evaluate:
+    parameters:
+      - $ref: '#/components/parameters/TpsObjectId'
+      - $ref: '#/components/parameters/Datacenter'
+      - $ref: '#/components/parameters/Platform'
+    get:
+      summary: Evaluate whether a datacenter is available inside of a region constraint
+      operationId: evaluateRegionDatacenter
+      tags: [Tps]
+      responses:
+        '200':
+          description: Constraint evaluated successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/TpsPaoGetResult'
```

Comment:
I don't think you want the `200` return here.

The spec suggests that the failure will return a list of data center ids. I do not see support for that in the code. How are you planning to convey those?

There is a `causes` array in the `ErrorReport` object that could be used. That would require the caller to understand how to split the string into platform/datacenter components to be usable. (If we expect the caller to do that, then maybe I was wrong about separating those fields in the API??)

I think we will need something like `GET /region/{objectid}/list-valid` or similar to produce the list of valid regions as well. An alternative to this approach would be to only provide that list of valid regions. WSM would need to ensure the incoming datacenter was on the list. But I like having both, I think.

Reply:
Added both of these in. There's one oddity it introduces in that if I leave the 204 with an empty response then openapi will generate a ResponseEntity<Void> and I can't construct the response list in the error case.

The ErrorReport also didn't have a causes array on it. Is that something you're adding?

Reply:
That is because I made some cut'n'paste error. If you look in WSM, you will see it has the properly defined error report schema [here](https://github.com/DataBiosphere/terra-workspace-manager/blob/fd18e3809a1c93d163f2362606a01029b2192b06/openapi/src/common/schemas.yaml#L106). Maybe you can fix that?

The `ErrorReportException` Java class is defined in TCL [here](https://github.com/DataBiosphere/terra-common-lib/blob/84d0db138bd713a339391519686308e160b58187/src/main/java/bio/terra/common/exception/ErrorReportException.java#L15) in the common lib carries the array.

In WSM, we have used `204` in several places without problem. One example is [here](https://github.com/DataBiosphere/terra-workspace-manager/blob/fd18e3809a1c93d163f2362606a01029b2192b06/openapi/src/parts/workspace.yaml#L95). Maybe you can trace that back and see how it looks in the controller.

I also see that we are missing a global exception handler that will convert exceptions with `ErrorReportException` from into the API `ErrorReport` object.

So how about you populate the "cause" array in the exception with the allowed regions and I will fix ErrorReport and add the global exception handler that is missing.

Reply:
Made [PF-2266](https://broadworkbench.atlassian.net/browse/PF-2266) for the error report fixes.